index.js

```diff
@@ -205,6 +205,17 @@ function sendRequest(client, device, message, projectId, accessToken, doneCallba
             // Convert response body to JSON object
             let response = JSON.parse(data);
 
+            // Extract status code from JSON response object
+            let statusCode = response.statusCode ?? response.status;
+            // Status code found?
+            if (statusCode) {
+                // Server-side error?
+                if (statusCode >= 500) {
+                    // Retry request using same HTTP2 session in 10 seconds
+                    return setTimeout(() => { sendRequest.apply(this, args) }, 10 * 1000);
```

Comment:
Might be better to use the existing `errorHandler()` method instead of repeating the `setTimeout()` call, and we can also limit the amount of retries this way. Can you replace this line with `return errorHandler(new Error(statusCode + 'Internal Server Error'));`?